Source/SwiftLintBuiltInRules/Rules/Style/RedundantDiscardableLetRule.swift

```diff
@@ -100,4 +147,56 @@ private extension AccessorBlockSyntax {
         }
         return false
     }
+
+    var isPreviewProviderBody: Bool {
+        guard let binding = parent?.as(PatternBindingSyntax.self),
+              binding.pattern.as(IdentifierPatternSyntax.self)?.identifier.text == "previews",
+              let bindingList = binding.parent?.as(PatternBindingListSyntax.self),
+              let variableDecl = bindingList.parent?.as(VariableDeclSyntax.self) else {
+            return false
+        }
+
+        guard variableDecl.modifiers.contains(keyword: .static) &&
+                variableDecl.bindingSpecifier.tokenKind == .keyword(.var) else {
+            return false
+        }
+
+        if let type = binding.typeAnnotation?.type.as(SomeOrAnyTypeSyntax.self) {
```

Comment:
Maybe is a `guard` as well. Then you can even combine all guards into one.

Reply:
Combined into a single guard here: https://github.com/realm/SwiftLint/pull/6075/commits/b8e67bff3a616dfc8a9df6d65155074a66a4a2fe